### ISSUE TITLE
Question about the perl script wrapper "bowite2", line 228.

### ISSUE THREAD

**Issue by MapleHe** [BenLangmead/bowtie2#233]
Hi, contributors. Thank you for your work.

I've read the source code of bowtie2 wrapper script, and couldn't understand the point of [line 228](https://github.com/BenLangmead/bowtie2/blob/f5d794d7588a5ce4a7e735c42667be5abe0cdaf2/bowtie2#L228). Besides, I guessed that the arguments of ```[un|al][-conc]-[bz2|lz4]``` output will cause an error after this "if-condition" block, as the argument itself will be pushed into ```bt2w_args``` as an unpaired fastq file path in this block, and then into ```unp``` array in GetOption function, and finally into ```bt2_args```  in scripts following line 413, 441.

```perl
    if ($arg =~ /(bz2|lz4)$/) {
        push @bt2w_args, ("-U", $arg);
        $bt2_args[$i] = undef;
    }
```

```--un-conc-gz``` will work normally as suffix "gz" will not be matched.

Here is my test:

command line:

```shell
perl bowtie2 -x /path/to/index -1 /path/to/read_1.fq -2 /path/to/read_2.fq --un-conc-bz2 /path/to/output_unconc.fq.bz2 -S /path/to/sam_out.sam
```

error:

```
stat: No such file or directory
Warning: Could not open read file "--un-conc-bz2" for reading; skipping...
Error: No input read files were valid
(ERR): bowtie2-align exited with value 1
```

**Comment by ch4rr0** [comment]
Good catch! I fixed the regular expression responsible for the error. My initial intention with the block referenced was to catch read files compressed with either lz4 or bz2, so that the wrapper could first decompress them before passing them to the bowtie2 binary.

The fix will be included in the next release.

**Comment by MapleHe** [comment]
Thank you ~

**Comment by ch4rr0** [comment]
This change is available in v2.4.0 and later.
